.github/workflows/docker.yml

```diff
@@ -5,36 +5,53 @@ on:
     # Weekly, at 03:00 on Monday UTC time
     - cron: "0 3 * * 1"
   pull_request:
-    paths-ignore:
-      - ".github/ISSUE_TEMPLATE/**"
-      - "*.md"
-      - "binder/**"
-      - "docs/**"
-      - "examples/**"
-      - ".pre-commit-config.yaml"
-      - ".readthedocs.yaml"
+    paths:
+      - ".github/workflows/docker.yml"
+
+      - "all-spark-notebook/"
+      - "base-notebook/"
+      - "datascience-notebook/"
+      - "minimal-notebook/"
+      - "pyspark-notebook/"
+      - "r-notebook/"
+      - "scipy-notebook/"
+      - "tensorflow-notebook/"
+
+      - "tagging/"
+      - "test/"
+      - "conftest.py"
+      - "Makefile"
+      - "pytest.ini"
+      - "requirements-dev.txt"
   push:
     branches:
-      - master
       - main
-    paths-ignore:
-      - ".github/ISSUE_TEMPLATE/**"
-      - "*.md"
-      - "binder/**"
-      - "docs/**"
-      - "examples/**"
-      - ".pre-commit-config.yaml"
-      - ".readthedocs.yaml"
+      - master
+    paths:
+      - ".github/workflows/docker.yml"
+
+      - "all-spark-notebook/"
+      - "base-notebook/"
+      - "datascience-notebook/"
+      - "minimal-notebook/"
+      - "pyspark-notebook/"
+      - "r-notebook/"
+      - "scipy-notebook/"
+      - "tensorflow-notebook/"
+
+      - "tagging/"
+      - "test/"
+      - "conftest.py"
+      - "Makefile"
+      - "pytest.ini"
+      - "requirements-dev.txt"
 
 jobs:
   build-test-publish-images:
     name: Build, test, and publish Docker Images
     runs-on: ubuntu-latest
     permissions:
       contents: write
-    if: >
-      !contains(github.event.head_commit.message, 'ci skip') &&
```

Comment:
Automatic pushes do not need to be checked, because they won't trigger github workflows, as far as I know :)